Biology.java

```diff
@@ -0,0 +1,346 @@
+            }
+            this.epitopeSites = epitopeSitesSet;
+
+            // create a mapping of site # to matrix of vectors
+            // m = wild type amino acid
+            // n = mutant type amino acid
+            int matrixSize = Parameters.AlphabetType.AMINO_ACIDS.getValidCharacters().length();
+            Map<Integer, double[][][]> currentSiteMutationVectors  = new HashMap<>();
+
+            // sites
+            for (int nucleotideSiteNumber = 0; nucleotideSiteNumber < Parameters.startingSequence.length() / 3; nucleotideSiteNumber += 1) {
+                String currentStringOutputCSV = "mutation,r,theta\n";
+
+                double[][][] currentSiteMutationMatrix = new double[matrixSize][matrixSize][];
+
+                // rows of a single matrix
+                for (int wildTypeIndex = 0; wildTypeIndex < matrixSize; wildTypeIndex++) {
+                    // columns of a single matrix
+                    for (int mutationIndex = 0; mutationIndex < matrixSize; mutationIndex++) {
+                        if (mutationIndex < wildTypeIndex) { // update lower and upper triangle in this branch.
+                            // direction of mutation
+                            double theta;
+                            if (Parameters.mut2D) {
+                                theta = Random.nextDouble(0, 2 * Math.PI);
+                            } else {
+                                if (Random.nextBoolean(0.5)) {
+                                    theta = 0;
+                                } else {
+                                    theta = Math.PI;
+                                }
+                            }
+
+                            // size of mutation
+                            double r;
+                            double alpha;
+                            double beta;
+
+                            double meanStep;
+                            double sdStep;
+                            if (this.epitopeSites.contains(nucleotideSiteNumber)) {
+                                // epitope sites
+                                meanStep = Parameters.meanStepEpitope;
+                                sdStep = Parameters.sdStepEpitope;
+                            } else {
+                                // non-epitope sites
+                                meanStep = Parameters.meanStep;
+                                sdStep = Parameters.sdStep;
+                            }
+
+                            r = meanStep;
+                            if (!Parameters.fixedStep) {
+                                alpha = (meanStep * meanStep) / (sdStep * sdStep);
+                                beta = (sdStep * sdStep) / meanStep;
+                                r = Random.nextGamma(alpha, beta);
+                            }
+
+                            // create phenotype
```

Comment:
Could we say "create antigenic phenotype by computing the x and y coordinates (mutA and mutB) of the virus in antigenic space"?